added punnett_square fx.

MdgAki opened the pull request:
punnett_square function added along with predator_factory.h & .cpp files. Function allows sim_message to pass one predator's two genes of one genotype in. If the two genes are the same, then the first one is returned. If the genes are different (sS), then a 50% random selection of s or S is returned. The returned character can be set to the child predator.

Commit:
added punnett_square fx.

asqapro commented on sources/predator_factory.cpp:
```diff
@@ -0,0 +1,65 @@
+{
+    if (gene_one != gene_two)
+    {
+        srand(time(NULL));
```
It's a good idea to `srand(...)` one time, probably in `init_sim()`

MdgAki replied:
Agreed, the way it is now though, it isn't global to anything. So, I don't see too many problems with it. Other than here and in the seed.cpp, where else do we need a random?

Commit:
moved over srand to init_sim.
added namespace to clock.
clock namespace avoids conflict with time.h
Need time.h for random (srand)

Commit:
Merge branch 'master' into mod/predator_factory/punnett_squares/adding_start_to_factory

Commit:
Added sim_ns to a few places where clock was naked

MdgAki commented:
See commit messages for info about sim_ns::
